### PR TITLE
fix: incosistent storage size/data retention size name and format

### PR DESCRIPTION
**Pull request by JameelB**
## Description
The [getInstanceTypesByCloudProviderAndRegion](https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/getInstanceTypesByCloudProviderAndRegion) endpoint returns Kafka storage size information with the following name and format:
```
...
"max_data_retention_size": {
    "bytes": 1073741800000
},
```

This is not the same for the rest of our endpoints that has this information in the response. See example:
```
{
  "id": "dummy-kafka-id",
  "kafka_storage_size": "10Gi"
  ...
}
```

The way this information is returned should be consistent throughout our public and private api. 

The changes in this PR results in the following:
- `max_data_retention_size` has now been added to the response of the following public and private admin API endpoints:
  - GET /kafkas
  - POST /kafkas
  - GET /kafkas/{id}
  - PATCH /kafkas/{id})
  - GET /admin/kafkas
  - GET /admin/kafkas/{id}
  - PATCH /admin/kafkas/{id}
  - DELETE /admin/kafkas/{id}

  **Example response**
  ```
  {
    "id": "dummy-kafka-id",
    "max_data_retention_size": {
      "bytes": 1073741800000
    },
    ...
  }
  ```
- `max_data_retention_size` has been added to the payload of the following endpoint:
  - PATCH /admin/kafkas/{id}

  **Example request**
  ```
  {
    "max_data_retention_size": "10Gi"
  }
  ```
- `kafka_storage_size` is now deprecated, `max_data_retention_size` should be used instead.

## Verification Steps

Related JIRA: [MGDSTRM-8442](https://issues.redhat.com/browse/MGDSTRM-8442)

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [x] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] JIRA has been created for changes required on the client side
- [ ] ~~Required metrics/dashboards/alerts have been added (or PR created).~~
- [ ] ~~Required Standard Operating Procedure (SOP) is added.~~

### CODE REVIEW

**Comment by machi1990** [comment]
> max_data_retention_size has been added to the payload of the following endpoint:

> PATCH /admin/kafkas/{id}
> Example request

> {
>  "max_data_retention_size": "10Gi"
> }

We might need to update the upgrade pipeline to take this into account

**Review by machi1990 — APPROVED** [review]
lgtm

pinging @wtrocki @riccardo-forina for awareness

**Comment by riccardo-forina** [comment]
How do we plan to handle the transition? A synchronized change isn't possible due to the distributed nature of the web apps, we can't be sure all clients will be instantly using the latest UI.

One option could be to release an updated SDK that removes the old field and adds the new one while handling both options under the hood. It will have to convert the K8s size string to bytes but it can be done with the same "logic" we use in the UI (which is stripping the `G` and multiplying the number for some constant value). The benefit of doing this is that it's a single point that contains the backward compatibility hack, that can easily be removed at a later moment.

File: internal/kafka/internal/api/admin/private/model_error.go
Comment (JameelB) [reply internal/kafka/internal/api/admin/private/model_error.go:6]: i'm ok with removing that comment. I don't think we'll be missing out on anything if it's gone.

**Comment by JameelB** [comment]
> How do we plan to handle the transition? A synchronized change isn't possible due to the distributed nature of the web apps, we can't be sure all clients will be instantly using the latest UI.
> 
> One option could be to release an updated SDK that removes the old field and adds the new one while handling both options under the hood. It will have to convert the K8s size string to bytes but it can be done with the same "logic" we use in the UI (which is stripping the `G` and multiplying the number for some constant value). The benefit of doing this is that it's a single point that contains the backward compatibility hack, that can easily be removed at a later moment.

Maybe @wtrocki would be best to answer this? :) As for the KAS Fleet Manager API, we will be returning both fields for awhile until the UI has fully migrated to use the new field.

**Comment by wtrocki** [comment]
I think @JameelB you have answered it very well. All fields are returned for both new and old format so no one (clients) would complain.

**Comment by wtrocki** [comment]
Verified with cli all good.

@dimakis we need to merge js-sdk PRs once this pr is merged. Best to have UI adopting this changes asao

**Comment by JameelB** [comment]
Thanks @machi1990 @wtrocki @riccardo-forina for the review! I'll merge this once the CI checks passes

**Pull request merged**
